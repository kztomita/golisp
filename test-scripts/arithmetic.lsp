(print (+))
(print (+ 1))
(print (+ 1 2))
(print (+ 1 2 3))
(print (+ 1.1 2))
(print (+ 1 2.1))
(print (+ 1.1 2.1))

(print "----")
(print (- 1))
(print (- 1 2))
(print (- 1 2 3))
(print (- 1 0.1))

(print "----")
(print (*))
(print (* 2))
(print (* 2 3))
(print (* 2 3 4))
(print (* 2 3.3))

(print "----")
(print (/ 2.0))
(print (/ 4 2))
(print (/ 4 3)); Common Lispでは分数4/3だが、本実装では整数型の切り捨てで1となる
(print (/ 100 2 5))
(print (/ 3 4.0))

(print "----")
(print (+ 4 (* 3 4)))
