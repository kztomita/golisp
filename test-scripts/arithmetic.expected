0
1
3
6
3.1
3.1
3.2
"----"
-1
-1
-4
0.9
"----"
1
2
6
24
6.6
"----"
0.5
2
1
10
0.75
"----"
16
